Sources/DequeModule/Deque+Equatable.swift

```diff
@@ -17,6 +17,16 @@ extension Deque: Equatable where Element: Equatable {
   /// - Complexity: O(`min(left.count, right.count)`)
   @inlinable
   public static func ==(left: Self, right: Self) -> Bool {
+    let lhsCount = left.count
+    if lhsCount != right.count {
+      return false
+    }
+
+    // Test referential equality.
+    if lhsCount == 0 || left._storage._buffer.buffer === right._storage._buffer.buffer {
```

Comment:
Super minor nit: We usually try to avoid directly accessing low-level layers within the implementation of high-level public API like this `==` implementation. It could be worth adding an `isIdentical(to:)` method to `Deque._Storage`, and then calling it here, instead of directly extracting the buffer references.